### ISSUE TITLE
vs. Tune Trainable API?

### ISSUE THREAD

**Issue by drozzy** [PyTorchLightning/pytorch-lightning#1471]
Hi guys.

How is this library different from Tune Trainable?
https://ray.readthedocs.io/en/latest/tune-usage.html

It seems like they offer the same ability to distribute computation, early stopping, checkpointing and even hyperparameter search, etc...

Just looking for some comments. Thanks.

**Comment by williamFalcon** [comment]
many ways!

Lightning is a platform... this means we integrate with any other AI related project. 

1. If you look through the issues we are adding support for Ray distributed in lightning for those who want to use ray (ie: tune, etc can add features to lightning without getting you locked into any particular approach. 
2. we support other approaches for hyperparameter search like tune, hydra, optuna, etc...
3. we’re a research platform, not just a trainer. ie: we also allow TPU training, 16 bit precision, automatic research tricks like backpropagation through time, etc...
4. we are just a way to organize pytorch code, so there’s no new framework to learn or anything. 

But overall, you can think of ray as a way to do hyperparameter search and another way of doing distributed (ie: gradient syncing, etc).  In lightning we support dp, ddp, ddp2 and soon ray. To use any of these you don’t  have to change your lightning code:

Trainer(distributed_backend="ray|ddp|dp”)
